File: Scripts/laptop/numerical-integration/javascript/numerical-integration.js
function integrateRectagle(func, a, b, n, k) {
	var h = (b-a)/n;
	var sum = 0;
	for (var i = a, j=a; i < b; i+=(b/n), ++j) {
		if (func == "cube") {
			sum += cube(i-h*(1-k));
		}
		
		if (func == "reciprocal") {
			sum += reciprocal(i-h*(1-k));
		}

		if (func == "identity") {
			sum += identity(i-h*(1-k));
		} 
	}
	sum *=  h;
}

function integrateTrapezoid(func, a, b, n) {
	var h = (b-a)/n;
	var sum = 0;
	for (var i = a, j=a; i < b; i+=(b/n), ++j) {
		if (func == "cube") {
			if (i < b-1) {
				sum += cube(i) + cube(i+1); 
			} else {
				sum += cube(i) + cube(i+1-b);
			}
		} 
		if (func == "reciprocal") {
			if (i < b-1) {
				sum += reciprocal(i) + reciprocal(i+1); 
			} else {
				sum += reciprocal(i) + reciprocal(i+1-b);
			}
		} 
		if (func == "identity") {
			if (i < b-1) {
				sum += identity(i) + identity(i+1); 
			} else {
				sum += identity(i) + identity(i+1-b);
			}
		} 
	}
	sum *= h/2;
}

function integrateSimpsons(func, a, b, n) {
	var h = (b-a)/h;
	var sum = 0;
	for (var i = a, j=a; i < b; i+=(b/n), ++j) {
		if (func == "cube") {
			if (i < b -1) {
				sum += cube(i) + (4 * cube(i - (h / 2))) + cube(i + 1);
			} else {
				sum += cube(i) + (4 * cube(i - (h / 2))) + cube(i + 1 - b);
			}
		}

		if (func == "reciprocal") {
			if (i < b -1) {
				sum += reciprocal(i) + (4 * reciprocal(i - (h / 2))) + reciprocal(i + 1);
			} else {
				sum += reciprocal(i) + (4 * reciprocal(i - (h / 2))) + reciprocal(i + 1 - b);
			}
		}
		if (func == "identity") {
			if (i < b -1) {
				sum += identity(i) + (4 * identity(i - (h / 2))) + identity(i + 1);
			} else {
				sum += identity(i) + (4 * identity(i - (h / 2))) + identity(i + 1 - b);
			}
		}
	}
        sum *= h/6;
}

function cube(x) {
	return (x*x*x+aVar);
}

function reciprocal(x) {
	return 1/(x+aVar);
}

function identity(x) {
	return x+aVar;
}

function executeTask(i) {

integrateRectagle("cube",0,1,100,0);
integrateRectagle("cube",0,1,100,0.5);
integrateRectagle("cube",0,1,100,1);
integrateTrapezoid("cube",0,1,100);
integrateSimpsons("cube",0,1,100);

integrateRectagle("reciprocal",1,100,1000,0);
integrateRectagle("reciprocal",1,100,1000,0.5);
integrateRectagle("reciprocal",1,100,1000,1);
integrateTrapezoid("reciprocal",1,100,1000);
integrateSimpsons("reciprocal",1,100,1000);

integrateRectagle("identity",0,5000,5000000,0);
integrateRectagle("identity",0,5000,5000000,0.5);
integrateRectagle("identity",0,5000,5000000,1);
integrateTrapezoid("identity",0,5000,5000000);
integrateSimpsons("identity",0,5000,5000000);

integrateRectagle("identity",0,6000,6000000,0);
integrateRectagle("identity",0,6000,6000000,0.5);
integrateRectagle("identity",0,6000,6000000,1);
integrateTrapezoid("identity",0,6000,6000000);
integrateSimpsons("identity",0,6000,6000000);
	return i;
}

var r = 1;
var aVar = 0;


for ( aVar = 0; aVar < 100; ++aVar) {
	r = executeTask(aVar);
}

File: scripts/task_tested_with_compile_script/numerical-integration/python/numerical-integration.py
from fractions import Fraction

aVar=0

def left_rect(f,x,h):
  return f(x)

def mid_rect(f,x,h):
  return f(x + h/2)

def right_rect(f,x,h):
  return f(x+h)

def trapezium(f,x,h):
  return (f(x) + f(x+h))/2.0

def simpson(f,x,h):
  return (f(x) + 4*f(x + h/2) + f(x+h))/6.0

def cube(x):
  return (x*x*x+aVar)

def reciprocal(x):
  return 1/(x+aVar)

def identity(x):
  return x+aVar

def integrate( f, a, b, steps, meth):
   h = (b-a)/steps
   ival = h * sum(meth(f, a+i*h, h) for i in range(steps))
   return ival

# Tests
r = 1

def executeTask(i):
  integrate(cube,0,1,100,left_rect)
  integrate(cube,0,1,100,mid_rect)
  integrate(cube,0,1,100,right_rect)
  integrate(cube,0,1,100,trapezium)
  integrate(cube,0,1,100,simpson)

  integrate(reciprocal,1,100,1000,left_rect)
  integrate(reciprocal,1,100,1000,mid_rect)
  integrate(reciprocal,1,100,1000,right_rect)
  integrate(reciprocal,1,100,1000,trapezium)
  integrate(reciprocal,1,100,1000,simpson)

  integrate(identity,0,5000,5000000,left_rect)
  integrate(identity,0,5000,5000000,mid_rect)
  integrate(identity,0,5000,5000000,right_rect)
  integrate(identity,0,5000,5000000,trapezium)
  integrate(identity,0,5000,5000000,simpson)

  integrate(identity,0,6000,6000000,left_rect)
  integrate(identity,0,6000,6000000,mid_rect)
  integrate(identity,0,6000,6000000,right_rect)
  integrate(identity,0,6000,6000000,trapezium)
  integrate(identity,0,6000,6000000,simpson)

  return i + 1

for aVar in range(100):
	r = executeTask(aVar)
